src/core/metadata/qgslayermetadata.h

```diff
@@ -322,6 +323,24 @@ class CORE_EXPORT QgsLayerMetadata : public QgsAbstractMetadataBase
 
     bool operator==( const QgsLayerMetadata &metadataOther ) const;
 
+    /**
+     * Returns TRUE if the metadata identifier, title, abstract, keywords or categories
+     * contain \a searchString using case-insensitive search.
+     *
+     * If \a searchString is empty this method returns TRUE.
```

Comment:
I'm still hesitant regarding this -- I think this should return false instead.

Reply:
ok, I'll look to it.

Reply:
done